pkg/core/xds/types.go

```diff
@@ -141,12 +143,113 @@ type Proxy struct {
 	Routing     Routing
 	Policies    MatchedPolicies
 
+	SecretsTracker SecretsTracker
+
 	// ZoneEgressProxy is available only when XDS is generated for ZoneEgress data plane proxy.
 	ZoneEgressProxy *ZoneEgressProxy
 	// ZoneIngressProxy is available only when XDS is generated for ZoneIngress data plane proxy.
 	ZoneIngressProxy *ZoneIngressProxy
 }
 
+type identityCertRequest struct {
+	meshName string
+}
+
+func (r identityCertRequest) Name() string {
+	return names.GetSecretName(xds_tls.IdentityCertResource, "secret", r.meshName)
+}
+
+type IdentityCertRequest interface {
+	Name() string
+}
+
+type CaRequest interface {
+	MeshName() []string
+	Name() string
+}
+
+type caRequest struct {
+	meshName string
+}
+
+type allInOneCaRequest struct {
+	meshNames []string
+}
+
+func (r caRequest) Name() string {
+	return names.GetSecretName(xds_tls.MeshCaResource, "secret", r.meshName)
+}
+
+func (r caRequest) MeshName() []string {
+	return []string{r.meshName}
+}
+
+func (r allInOneCaRequest) Name() string {
+	return names.GetSecretName(xds_tls.MeshCaResource, "secret", "all")
+}
+
+func (r allInOneCaRequest) MeshName() []string {
+	return r.meshNames
+}
+
+type SecretsTracker interface {
```

Comment:
Could you maybe comment why we need this? That this solves a problem that we need to deliver exactly the secrets that needs to be delivered.